core-processor/src/main/java/io/micronaut/expressions/parser/ast/operator/binary/MatchesOperator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.expressions.parser.ast.operator.binary;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.expressions.parser.ast.ExpressionNode;
+import io.micronaut.expressions.parser.ast.literal.StringLiteral;
+import io.micronaut.expressions.parser.compilation.ExpressionCompilationContext;
+import io.micronaut.expressions.parser.exception.ExpressionCompilationException;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.commons.GeneratorAdapter;
+import org.objectweb.asm.commons.Method;
+
+import static io.micronaut.expressions.parser.ast.types.TypeDescriptors.BOOLEAN;
+import static io.micronaut.expressions.parser.ast.types.TypeDescriptors.STRING;
+
+/**
+ * Expression AST node for regex 'matches' operator.
+ *
+ * @author Sergey Gavrilov
+ * @since 4.0.0
+ */
+@Internal
+public class MatchesOperator extends ExpressionNode {
+    private final ExpressionNode operand;
+    private final StringLiteral pattern;
+
+    public MatchesOperator(ExpressionNode operand, StringLiteral pattern) {
```

Comment:
can we check if the pattern is a correct regex pattern and fail compilation otherwise?